src/SmartTransactionsController.ts

```diff
@@ -970,6 +1015,25 @@ export default class SmartTransactionsController extends StaticIntervalPollingCo
     return this.#chainId;
   }
 
+  #getChainIds(): Hex[] {
+    const { networkConfigurationsByChainId } = this.messagingSystem.call(
+      'NetworkController:getState',
+    );
+    return Object.keys(networkConfigurationsByChainId).filter(
+      (chainId): chainId is Hex =>
+        this.#supportedChainIds.includes(chainId as Hex),
+    );
+  }
+
+  #getNetworkClientId({ chainId }: { chainId: string }): string {
+    const { networkConfigurationsByChainId } = this.messagingSystem.call(
+      'NetworkController:getState',
+    );
+    return networkConfigurationsByChainId[chainId as Hex].rpcEndpoints[
+      networkConfigurationsByChainId[chainId as Hex].defaultRpcEndpointIndex
```

Comment:
Are these things always present, e.g. `networkConfigurationsByChainId[chainId as Hex].defaultRpcEndpointIndex`?

Reply:
yes the value inside networkConfigurationsByChainId are referring to the networkController state and always present